cmd/notation/blob/inspect.go

```diff
@@ -0,0 +1,157 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+import (
+	"errors"
+	"fmt"
+	"github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation-go/plugin/proto"
+	"github.com/notaryproject/notation-go/registry"
+	"github.com/notaryproject/notation/cmd/notation/internal/sharedutils"
+	"github.com/notaryproject/notation/internal/cmd"
+	"github.com/notaryproject/notation/internal/envelope"
+	"github.com/notaryproject/notation/internal/ioutil"
+	"github.com/notaryproject/notation/internal/tree"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"strconv"
+)
+
+type blobInspectOpts struct {
+	cmd.LoggingFlagOpts
+	cmd.SignerFlagOpts
+	desc          ocispec.Descriptor
+	sigRepo       registry.Repository
+	signaturePath string
+	outputFormat  string
+	MediaType     string
+}
+
+func inspectCommand(opts *blobInspectOpts) *cobra.Command {
+	if opts == nil {
+		opts = &blobInspectOpts{}
+	}
+	longMessage := `Inspect all signatures associated with the signed artifact.
+
+Example - Inspect signatures on an BLOB artifact:
+  notation blob inspect <signature_path>
+
+Example - Inspect signatures on an BLOB artifact output as json:
+  notation blob inspect --output json <signature_path>
+`
+
+	command := &cobra.Command{
+		Use:   "blob inspect [signaturePath]",
+		Short: "Inspect all signatures associated with the signed artifact",
+		Long:  longMessage,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("missing signature path to the artifact: use `notation blob inspect --help` to see what parameters are required")
+			}
+			opts.signaturePath = args[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runBlobInspect(cmd, opts)
+		},
+	}
+
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	cmd.SetPflagOutput(command.Flags(), &opts.outputFormat, cmd.PflagOutputUsage)
+	return command
+}
+
+func runBlobInspect(command *cobra.Command, opts *blobInspectOpts) error {
+	// set log level
+	ctx := opts.LoggingFlagOpts.InitializeLogger(command.Context())
+
+	if opts.outputFormat != cmd.OutputJSON && opts.outputFormat != cmd.OutputPlaintext {
+		return fmt.Errorf("unrecognized output format %s", opts.outputFormat)
+	}
+
+	output := sharedutils.InspectOutput{MediaType: opts.MediaType, Signatures: []sharedutils.SignatureOutput{}}
+	//Added sigRepo and desc as placeholders. Once notation-go changes are merged revisit and update FetchSignatureBlob
```

Comment:
There are no changes innotation-go for blob inspect command.

Reply:
Took a note and removed the comment